Source/WebKit/WebProcess/GPU/webrtc/LibWebRTCCodecs.cpp

```diff
@@ -70,45 +66,59 @@ static webrtc::WebKitVideoDecoder createVideoDecoder(const webrtc::SdpVideoForma
     if (equalIgnoringASCIICase(codecString, "H264"_s))
         return { codecs.createDecoder(VideoCodecType::H264), false };
 
-    if (equalIgnoringASCIICase(codecString, "H265"_s))
-        return { codecs.createDecoder(VideoCodecType::H265), false };
-
     if (equalIgnoringASCIICase(codecString, "VP9"_s) && codecs.supportVP9VTB())
         return { codecs.createDecoder(VideoCodecType::VP9), false };
 
     if (equalIgnoringASCIICase(codecString, "AV1"_s)) {
+        if (codecs.hasAV1HardwareDecoder())
+            return { codecs.createDecoder(VideoCodecType::AV1), false };
         auto av1Decoder = createLibWebRTCDav1dDecoder().moveToUniquePtr();
         return { av1Decoder.release(), true };
     }
 
+    if (equalIgnoringASCIICase(codecString, "H265"_s))
+        return { codecs.createDecoder(VideoCodecType::H265), false };
+
     return { };
 }
 
 std::optional<VideoCodecType> LibWebRTCCodecs::videoCodecTypeFromWebCodec(const String& codec)
 {
-    // WebCodecs is assuming case sensitive comparisons.
+    if (codec.startsWith("avc1."_s))
+        return VideoCodecType::H264;
+
     if (codec.startsWith("vp09.0"_s)) {
         if (!supportVP9VTB())
             return { };
         return VideoCodecType::VP9;
     }
 
-    if (codec.startsWith("avc1."_s))
-        return VideoCodecType::H264;
+    if (codec.startsWith("av01."_s)) {
+        if (!hasAV1HardwareDecoder())
+            return { };
+        return VideoCodecType::AV1;
+    }
 
     if (codec.startsWith("hev1."_s) || codec.startsWith("hvc1."_s))
         return VideoCodecType::H265;
 
     return { };
 }
 
-void LibWebRTCCodecs::setHasVP9ExtensionSupport(bool hasVP9ExtensionSupport)
+void LibWebRTCCodecs::setHasAV1HardwareDecoder(bool hasAV1HardwareDecoder)
 {
-    m_hasVP9ExtensionSupport = hasVP9ExtensionSupport;
+    m_hasAV1HardwareDecoder = hasAV1HardwareDecoder;
+    if (!hasAV1HardwareDecoder)
+        return;
+    Page::forEachPage([](auto& page) {
```

Comment:
shouldn't this be using a PlatformStrategy instead (a singleton)?

Reply:
We can rework it in a follow-up if we think this is useful.